File: src/main/java/jisop/domain/DefaultTypeConverter.java
package jisop.domain;

import java.util.function.BiFunction;

/**
 * Created by mathieu.
 */
public class DefaultTypeConverter implements BiFunction<Class,String,Object>
{
    public Object apply(Class cls, String value) {
        if (cls==String.class){
            return value;
        }
        if (cls==Integer.class){
            return Integer.parseInt(value);
        }
        if (cls==int.class){
            return Integer.parseInt(value);
        }
        if (cls==double.class){
            return Double.parseDouble(value);
        }
        throw new RuntimeException("not implemented "+cls.getName());
    }
}

File: src/main/java/isop/domain/DefaultTypeConverter.kt
package isop.domain

import java.util.function.BiFunction

/**
 * Created by mathieu.
 */
class DefaultTypeConverter : BiFunction<Class<*>, String, Any> {
    override fun apply(cls: Class<*>, value: String): Any {
        if (cls == String::class.java) {
            return value
        }
        if (cls == Int::class.java) {
            return Integer.parseInt(value)
        }
        if (cls == Int::class.javaPrimitiveType) {
            return Integer.parseInt(value)
        }
        if (cls == Double::class.javaPrimitiveType) {
            return java.lang.Double.parseDouble(value)
        }
        throw RuntimeException("not implemented " + cls.name)
    }
}